src/lib.rs

```diff
@@ -8,163 +8,154 @@
 
-    // Load the workspace and current package
-    let workspace = workspace(&config, None)?;
-    let package = workspace.current()?;
+    let resolve = metadata
+        .resolve
+        .ok_or_else(|| format_err!("No project dependences"))?
+        .nodes;
 
-    // Resolve all dependencies (generate or use Cargo.lock as necessary)
-    let mut registry = registry(&config, &package)?;
-    let resolve = resolve(&mut registry, &workspace)?;
+    let mut git_crates = Vec::new();
 
     // build the crates the package needs
     let mut crates = resolve
-        .1
         .iter()
-        .map(|pkg| format!("{}-{}\n", pkg.name(), pkg.version()))
+        .cloned()
+        .filter_map(|pkg| {
+            let infopkg: Vec<&str> = pkg.id.split(' ').collect();
+            if infopkg.len() != 3 {
+                None
+            } else if infopkg[2].starts_with("(git") {
+                git_crates.push(infopkg[2][1..infopkg[2].len()].to_string());
+                None
+            } else {
+                Some(format!("{}-{}", infopkg[0], infopkg[1]))
+            }
+        })
         .collect::<Vec<String>>();
 
     // sort the crates
     crates.sort();
 
-    // root package metadata
-    let metadata = package.manifest().metadata();
-
-    // package description
-    let desc = metadata
-        .description
-        .as_ref()
-        .cloned()
-        .unwrap_or_else(|| String::from(package.name()));
-
-    // package homepage
-    let homepage = metadata.homepage.as_ref().cloned().unwrap_or(
-        metadata
-            .repository
-            .as_ref()
-            .cloned()
-            .unwrap_or_else(|| String::from("")),
-    );
-
-    let license = metadata
-        .license
-        .as_ref()
-        .cloned()
-        .unwrap_or_else(|| String::from("unknown license"));
+    let mut string = String::new();
+    File::open(&manifest)?.read_to_string(&mut string)?;
+
+    let parsed_manifest = string.parse::<Value>()?;
+    let table = &parsed_manifest
+        .as_table()
+        .ok_or_else(|| format_err!("Cargo manifest does not contain a toml table"))?;
+    let package = table
+        .get("package")
+        .ok_or_else(|| format_err!("Field \"package\" is missing in Cargo manifest"))?;
+    let name = read_from_manifest(package, &"name").unwrap_or_else(|| String::from(""));
+    let license = read_from_manifest(package, &"license").unwrap_or_else(|| String::from(""));
+    let description = read_from_manifest(package, &"description").unwrap_or_else(|| name.clone());
+    let homepage = read_from_manifest(package, &"homepage").unwrap_or_else(|| String::from(""));
+    let version = read_from_manifest(package, &"version").unwrap_or_else(|| String::from(""));
 
     // build up the ebuild path
-    let ebuild_path = PathBuf::from(format!("{}-{}.ebuild", package.name(), package.version()));
+    let path = PathBuf::from(ebuild_path.unwrap_or_else(|| format!("{}-{}.ebuild", name, version)));
```

Comment:
Wouldn't a better fallback be to the current directory so that the code below could be used as the fallback. Instead there's two failure cases to handle. I would clean this up.